src/plib/library/Task/Scan.php

```diff
@@ -17,29 +17,44 @@ public function run()
             $result = \pm_ApiCli::callSbin('diskspace_usage.sh', [$path, $this->getParam('username')]);
         }
 
+        $fileManager = $this->getParam('isAdmin') ? new \pm_ServerFileManager : new \pm_FileManager(\pm_Session::getCurrentDomain()->getId());
```

Comment:
do not rely on pm_Session in long task!
pass domain ID in parameter (instead of username)